core/ir/synthetic/encoding_common.h

```diff
@@ -33,8 +33,84 @@
 #ifndef _SYNTHETIC_ENCODING_COMMON_H_
 #define _SYNTHETIC_ENCODING_COMMON_H_
 
+/* This synthetic ISA is a made up ISA that has the purpose of preserving register
+ * dependencies and giving hints on the type of operations each instruction is performing.
+ * For this reason the majority of operations that would normally work on instructions
+ * coming from an actual ISA are not supported.
+ * The only operations that we support are instr_encode() and decode() when DynamoRIO's
+ * host is an x86 architecture.
```

Comment:
See other comment: it looks like encoding will work for any build not just x86.

Reply:
Yes. https://github.com/DynamoRIO/dynamorio/pull/6691#discussion_r1533506865